instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/middleware.py

```diff
@@ -101,21 +101,23 @@ def process_request(self, request):
 
         tracer = get_tracer(__name__, __version__)
 
-        attributes = collect_request_attributes(environ)
-        for attr in self._traced_request_attrs:
-            value = getattr(request, attr, None)
-            if value is not None:
-                attributes[attr] = str(value)
-
         span = tracer.start_span(
             self._get_span_name(request),
             kind=SpanKind.SERVER,
-            attributes=attributes,
             start_time=environ.get(
                 "opentelemetry-instrumentor-django.starttime_key"
             ),
         )
 
+        if span.is_recording():
+            attributes = collect_request_attributes(environ)
+            for attr in self._traced_request_attrs:
+                value = getattr(request, attr, None)
+                if value is not None:
+                    attributes[attr] = str(value)
```

Comment:
@owais created [a nifty utils function](https://github.com/open-telemetry/opentelemetry-python/pull/1178/files#diff-2be66a969f6d05eacdb47b93b9e8c5faR22) that you can use [like this](https://github.com/open-telemetry/opentelemetry-python/pull/1178/files#diff-ac6dff41851d3dc5dde99fdf324ff694R106)

```suggestion
            attributes = collect_request_attributes(environ)
            attributes = extract_attributes_from_object(
                request, self._traced_request_attrs, attributes
            )
```

It's **still in PR** though so no need to use it if this is ready first.